### DISCUSSION TITLE
Does our app meet the qualifications for being a formula?

### DISCUSSION THREAD

**Discussion by connor-oleary-pieces** [Homebrew/discussions#3657] [Writing Formulae/Casks] [answered]
Hello!

Hoping that this is the correct place to ask this. The company that i work for, [Pieces](https://code.pieces.app), has had me working on a CLI application to go along with our suite of tools (this post is not meant to advertise us, I swear!) and they would like me to PR it against homebrew core.

I was reading the Homebrew Documentation, specifically the Acceptable Formulae docs, and noticed that our application hits a few red flags but the end of the doc has a "Sometimes there are exceptions" disclaimer, so I'll list out some of the things I think might be wrong and I hope to gather some opinions on this.

For context, the CLI is meant for saving code snippets, either via selecting files or for saving recently run commands on the terminal (MacOS only right now). In order to use most of the functionality, our CLI requires the download of an additional desktop application that we call Pieces OS. This server contains some ML models that classify, tag, describe and save the snippet. We also have 1 feature that calls out to our cloud servers for sharing snippets with generated links.

Now for the actual discussion, I'll be listing the different areas I think we might be flagged for:
1. [Not Depending on proprietary software](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae)
As mentioned in the context section, the CLI depends on Pieces OS to function, and Pieces OS is proprietary. I know that core is also only hosting open-source code. While we could make the CLI open-source, the OS server would not be.
2. [Niche or self-submitted stuff](https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff)
I would be submitting my own code. We're a small startup, so we're definitely not well known and the CLI is currently not heavily used. It is maintained by me, however, and will continue to see regular updates. The only part that I think would qualify as "obscure" would be the server it depends on.
3. [Post-install intervention](https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-heavy-manual-prepost-install-intervention)
Again, back to the requirement for Pieces OS, I believe that the requirement to go to our site and download it would violate this section as well.

Thanks in advance for any advice/help!

**Comment by carlocab** [comment]
Depending on proprietary software is a deal-breaker, I'm afraid. That's something we do not make exceptions for at all.

You can always host this formula in your own tap. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Comment by SMillerDev** [comment, marked as answer]
If the CLI is open source and compiles without the server. That's fine. We host open source software that connects to proprietary software. The problem would be if it needs the server installed to compile the cli.

**Reply by connor-oleary-pieces** [reply]
It does not need the server to compile. The server just needs to be installed after installing the CLI to do the commands. I'll open a PR soon against core once I have the repo up open source.
